### PR TITLE
Checkout billing

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

**Comment by lderrickable** [comment]
This pull request is dependent on https://github.com/graycoreio/daffodil/pull/32

File: apps/foundation-demo/src/app/checkout/components/payment/billing-summary/billing-summary.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div class="billing-summary">
+  <div class="billing-summary__title">Billing Address</div>
+  <address-summary *ngIf="(billingAddress && !billingAddressIsShippingAddress)" [address]="billingAddress"></address-summary>
+  <div class="billing-summary__billing-address-is-shipping-address" *ngIf="billingAddressIsShippingAddress">Billing address is shipping address</div>
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/checkout/components/payment/billing-summary/billing-summary.component.html]: This classname is silly. Let's generalize it a bit more.

Comment (lderrickable) [reply apps/foundation-demo/src/app/checkout/components/payment/billing-summary/billing-summary.component.html]: renamed to billing-summary__note

**Comment by lderrickable** [comment]
Just in response to your comment. I did try to separate them at first. I actually spent a lot of time on that. But what eventually made me decide to leave them together is because of how the payment/billing form is designed. Instead of being two forms, it's one, and forms don't like being separated very much. I think what I would have needed to do to make them two separate modules wouldn't have been worth it.

**Pull request merged**